crates/router/src/utils/user.rs

```diff
@@ -207,16 +207,62 @@ pub fn get_redis_connection(state: &SessionState) -> UserResult<Arc<RedisConnect
         .attach_printable("Failed to get redis connection")
 }
 
-impl ForeignFrom<user_api::AuthConfig> for UserAuthType {
-    fn foreign_from(from: user_api::AuthConfig) -> Self {
+impl ForeignFrom<&user_api::AuthConfig> for UserAuthType {
+    fn foreign_from(from: &user_api::AuthConfig) -> Self {
         match from {
-            user_api::AuthConfig::OpenIdConnect { .. } => Self::OpenIdConnect,
-            user_api::AuthConfig::Password => Self::Password,
-            user_api::AuthConfig::MagicLink => Self::MagicLink,
+            &user_api::AuthConfig::OpenIdConnect { .. } => Self::OpenIdConnect,
+            &user_api::AuthConfig::Password => Self::Password,
+            &user_api::AuthConfig::MagicLink => Self::MagicLink,
         }
     }
 }
 
+pub async fn construct_public_and_private_db_configs(
+    auth_config: &user_api::AuthConfig,
+    encryption_key: &[u8],
+) -> UserResult<(Option<Encryption>, Option<serde_json::Value>)> {
+    match auth_config {
+        user_api::AuthConfig::OpenIdConnect {
+            private_config,
+            public_config,
+        } => {
+            let private_config_value = serde_json::to_value(private_config.clone())
+                .change_context(UserErrors::AuthConfigParsingError)
```

Comment:
why is this `400` ? isnt invalid value caught at api level ?
or someone can pass api model validation (serde) but still provide invalid json ?